File: module_common/src/main/java/module/common/utils/StringUtils.java
package module.common.utils;

import android.text.TextUtils;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.regex.Pattern;

/**
 * @describe: 字符串工具类
 * @date: 2019/9/2
 * @author: Mr Bai
 */
public class StringUtils {

    /**
     * @describe: 包装空字符串
    * @date:2019/9/2
     */
    public static String packNull(String content) {
        if(content==null){
            content = "";
        }
        return content;
    }

    public static boolean isPhoneNo(String phone) {
        String pattern = "^\\d{8}$|^1[\\d]{10}$";
        return Pattern.matches(pattern, phone);
    }

    @NotNull
    public static String format(String format,String number) {
        return  String.format(format, number);
    }

    @Nullable
    public static String phonePartHide(@Nullable String phone) {
        if(!TextUtils.isEmpty(phone)){
            String start = phone.substring(0, 3);
            String end = phone.substring(phone.length() - 4);
            return start+"***"+end;
        }
        return "";
    }
}

File: module_common/src/main/java/module/common/utils/StringUtils.kt
package module.common.utils

import android.text.TextUtils
import java.util.regex.Pattern

/**
 * @describe: 字符串工具类
 * @date: 2019/9/2
 * @author: Mr Bai
 */
object StringUtils {
    /**
     * @describe: 包装空字符串
     * @date:2019/9/2
     */
    fun packNull(content: String?): String {
        var content = content
        if (content == null) {
            content = ""
        }
        return content
    }

    fun isPhoneNo(phone: String?): Boolean {
        val pattern = "^\\d{8}$|^1[\\d]{10}$"
        return Pattern.matches(pattern, phone)
    }

    fun format(format: String?, number: String?): String {
        return String.format(format!!, number)
    }

    fun phonePartHide(phone: String?): String? {
        if (!TextUtils.isEmpty(phone)) {
            val start = phone!!.substring(0, 3)
            val end = phone.substring(phone.length - 4)
            return "$start***$end"
        }
        return ""
    }
}